src/main/java/com/cloudbees/hudson/plugins/folder/AbstractFolder.java

```diff
@@ -1092,13 +1159,39 @@ public void renameTo(String newName) throws IOException {
      * {@inheritDoc}
      */
     @Override
-    public synchronized void doSubmitDescription(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException {
+    public synchronized void doSubmitDescription(StaplerRequest2 req, StaplerResponse2 rsp) throws IOException, ServletException {
+        if (Util.isOverridden(AbstractFolder.class, getClass(), "doSubmitDescription", StaplerRequest.class, StaplerResponse.class)) {
```

Comment:
Most if not all of these web methods could probably be simplified considerably by just deleting the deprecated overload, which I doubt is either called programmatically or overridden by any known subclass. Pending that, it would make sense to mark the new overloads `final` and/or `@Restricted(NoExternalUse.class)` unless and until some legitimate use case for overriding one is identified.

(not referring to `submit`, which _is_ clearly overridden at least within this plugin, and also in `branch-api`)

Reply:
Does not seem necessary for this PR.